### PR TITLE
Bug/2218 user quota errors

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #2218

## Relevant technical choices

* The actual implementation is slightly different from the IB but it does the same.

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: includes/Modules/Analytics.php [hunk 1]
```diff
@@ -1220,21 +1220,40 @@ function ( Google_Service_Analytics_Account $account ) {
 					if ( in_array( $account_id, $account_ids, true ) ) {
 						$properties_profiles = $this->get_data( 'properties-profiles', array( 'accountID' => $account_id ) );
 					} else {
-						// Iterate over each account in reverse so if there is no match,
-						// the last $properties_profiles will be from the first account (selected by default).
-						foreach ( array_reverse( $accounts ) as $account ) {
+						$fallback_properties_profiles = null;
+						$accounts_array               = array_values( $accounts );
+
+						for ( $i = 0, $len = min( 25, count( $accounts_array ) ); $i < $len; $i++ ) {
```
Comment (tofumatt) [inline includes/Modules/Analytics.php:1226]: Does this need to be declared inside the `for` declaration? It's a bit harder to read this way and I'm not clear on why it needs to be in there.

(Also, `$length` or `$number_of_accounts` might be a more descriptive variable name.)

**Review by tofumatt — APPROVED** [review]
Looks good to me, but can you add a QA brief to the issue and then send it over to Merge Review? It's missing the QA brief but otherwise looks fine to me. 👍🏻

File: includes/Modules/Analytics.php (hunk 1 above)
Comment (eugene-manuilov) [reply includes/Modules/Analytics.php]: Updated.

**Review by tofumatt — APPROVED** [review]
Looks good, but the issue itself is still missing a QA brief.

This one could be tough to QA I suppose, but I think it's worth adding a QA brief or at least steps to test this new behaviour.

**Review by aaemnnosttv — COMMENTED** [review]
Thanks @eugene-manuilov – I found the logic a bit difficult to follow here so I created an alternate PR with my approach here: https://github.com/google/site-kit-wp/pull/2598. Would you please review it?

**Comment by eugene-manuilov** [comment]
@aaemnnosttv your PR looks good to me! :+1: We can close this one.

**Comment by aaemnnosttv** [comment]
Closing in favor of #2598

**Pull request closed without merging**